digitaltwin/modelrepo/Authentication/ModelSharedAccessSignature.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Microsoft.Azure.Devices.Common.Authorization
+{
+    public class ModelSharedAccessSignature : SharedAccessSignature
+    {
+        internal string _repositoryId;
+
+        private ModelSharedAccessSignature(string shareAccessSignatureName, DateTime expiresOn, string expiry, string keyName, string signature, string encodedAudience, string repositoryId) : base(shareAccessSignatureName, expiresOn, expiry, keyName, signature, encodedAudience)
+        {
+            _repositoryId = repositoryId;
+        }
+
+        public string RepositoryId
+        {
+            get
+            {
+                return _repositoryId;
+            }
+        }
+        public static ModelSharedAccessSignature ParseForModel(string shareAccessSignatureName, string rawToken)
+        {
+            if (string.IsNullOrWhiteSpace(shareAccessSignatureName))
+            {
+                throw new ArgumentNullException(nameof(shareAccessSignatureName));
+            }
+
+            if (string.IsNullOrWhiteSpace(rawToken))
+            {
+                throw new ArgumentNullException(nameof(rawToken));
+            }
+
+            IDictionary<string, string> parsedFields = ExtractFieldValues(rawToken);
+
+            string signature;
+            if (!parsedFields.TryGetValue(SharedAccessSignatureConstants.SignatureFieldName, out signature))
```

Comment:
more inline variable declarations possible in this file `parsedFields.TryGetValue(..., out string signature)`

This "new" syntax helps reduce lines of code that aren't very interesting. :)